core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java

```diff
@@ -619,6 +623,10 @@ public boolean escape() {
      * Default is <code>{@value #DEFAULT_SPLIT_ON_WHITESPACE}</code>.
      */
     public QueryStringQueryBuilder splitOnWhitespace(boolean value) {
+        if (autoGeneratePhraseQueries() && value == false) {
+            throw new IllegalArgumentException("it is disallowed to disable split_on_whitespace " +
+                "if auto_generate_phrase_queries is activated");
+        }
```

Comment:
Should we fail in `doToQuery` instead? I am wondering that the builder might be in an invalid state temporarily but in an ok state in the end depending on the order in which setters are called?